### PR TITLE
Consolidate interfaces into single package

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

As we're going to merge interfaces, `@unimodules/core` and `@unimodules/react-native-adapter` into single package, this is the first step in achieving this goal. 

# How

- Created new package `expo-modules-core` which will replace these three packages
- Copied interfaces into this package
- Android's package group `org.unimodules.interfaces` has been replaced with `expo.modules.interfaces`
- Replaced `UM` prefixes with `EX` on iOS

# To do later

Gradually migrate other packages to use interfaces from `expo-modules-core`

# Test Plan

Expo Go compiles

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-modules-core/android/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+<manifest package="expo.modules.core">
```
Comment (tsapeta) [inline packages/expo-modules-core/android/src/main/AndroidManifest.xml:2]:
```suggestion
<manifest package="expo.modules">
```

File: packages/expo-modules-core/android/build.gradle [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+android {
+  compileSdkVersion safeExtGet("compileSdkVersion", 30)
+
+  defaultConfig {
+    minSdkVersion safeExtGet("minSdkVersion", 21)
+    targetSdkVersion safeExtGet("targetSdkVersion", 30)
+    versionCode 15
+    versionName "6.1.0"
```
Comment (tsapeta) [inline packages/expo-modules-core/android/build.gradle:41]:
```suggestion
    versionCode 1
    versionName "0.0.1"
```

File: packages/expo-modules-core/android/build.gradle [hunk 3]
```diff
@@ -0,0 +1,46 @@
+apply plugin: 'com.android.library'
```
Comment (bbarthec) [inline packages/expo-modules-core/android/build.gradle:1]: I believe we need to add kotlin to this module: https://github.com/expo/expo/pull/12716

File: packages/expo-modules-core/android/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (bbarthec) [inline packages/expo-modules-core/android/src/main/AndroidManifest.xml:1]:
```suggestion
```

**Review by bbarthec — COMMENTED** [review]
I'm not really sure what's planned for this package in the future (these interfaces are going to be deleted completely? 🤔), but what would you say about rewriting all these Java files into Kotlin?

**Comment by tsapeta** [comment]
> I'm not really sure what's planned for this package in the future (these interfaces are going to be deleted completely? 🤔)

We haven't really made any decision about its future, but in my opinion
- `expo-modules-core` needs to be created, as a target for merged `@unimodules/core`, `@unimodules/react-native-adapter` and `react-native-unimodules`
- in the past we've had some discussions about deleting them... but why? if they are consolidated in one package, they don't bother anything as well

> what would you say about rewriting all these Java files into Kotlin?

Yes, sounds good, but first let's focus on the key point of this PR which is to create a new module and copy existing interfaces there 😛

**Review by bbarthec — APPROVED** [review]
@tsapeta, given your response, I can approve this one 👍 
Looking forward to simplifying this whole `interfaces-*`, `rn-adapter` and `core` thing 👀

**Pull request merged**